util/ui/ui.go

```diff
@@ -274,6 +276,27 @@ func (ui *UI) DisplayTextWithFlavor(template string, templateValues ...map[strin
 	fmt.Fprintf(ui.Out, "%s\n", ui.TranslateText(template, firstTemplateValues))
 }
 
+// DisplayJSON encodes and indents the input
+// and outputs the result to ui.Out.
+func (ui *UI) DisplayJSON(name string, jsonData interface{}) error {
+	buff := new(bytes.Buffer)
+	encoder := json.NewEncoder(buff)
+	encoder.SetEscapeHTML(false)
+	encoder.SetIndent("", " ")
+
+	err := encoder.Encode(jsonData)
+	if err != nil {
+		return err
+	}
+
+	if name != "" {
+		ui.DisplayText(fmt.Sprintf("%s: %s", name, buff))
```

Comment:
We have access to private methods that would be better than `DisplayText()`, for example:
```go
ui.terminalLock.Lock()
defer ui.terminalLock.Unlock()
fmt.Fprintf(...)
```